src/main/java/org/jenkinsci/test/acceptance/po/ArtifactManagement.java

```diff
@@ -45,12 +46,12 @@ public ArtifactManagement(JenkinsConfig config) {
     }
 
     public void clear() {
-        try {
-            while (true) {
+        while (control("artifactManagerFactories/repeatable-delete").exists()) {
```

Comment:
What the clear method does is to look for a possibly existing 'Delete' button, that is, the delete button for an item in the Artifact Management section, and clicks on it in order to remove the item. It does it iteratively until no more items (i.e. delete-buttons) are found, when the section is considered *cleared*.